demo/cpp/model_peeker.cc

```diff
@@ -0,0 +1,99 @@
+#include <iostream>
+#include <cstdio>
+#include <fstream>
+#include <numeric>
+#include <algorithm>
+#include <functional>
+#include <vector>
+#include <limits>
+#include <stdexcept>
+#include <dlr.h>
+#include "dmlc/logging.h"
+
+/*! \brief Prints model metadata using DLR C-API.
+ */
+void peek_model(DLRModelHandle model){
+  int num_inputs;
+  int num_weights;
+  int num_outputs;
+  const char* backend;
+  std::vector<const char*> input_names;
+  std::vector<const char*> weight_names;
+  std::vector<std::vector<int64_t>> output_shapes;
+
+  GetDLRBackend(&model, &backend);
+  std::cout << "backend is " << backend << std::endl;
+
+  GetDLRNumInputs(&model, &num_inputs);
+  std::cout << "num_inputs = " << num_inputs << std::endl;
+
+  GetDLRNumWeights(&model, &num_weights);
+  std::cout << "num_weights = " << num_weights << std::endl;
+
+  GetDLRNumOutputs(&model, &num_outputs);
+  std::cout << "num_outputs = " << num_outputs << std::endl;
+
+  input_names.resize(num_inputs);
+  std::cout << "input_names: ";
+  for (int i = 0; i < num_inputs; i++) {
+    GetDLRInputName(&model, i, &input_names[i]);
+    std::cout << input_names[i] << ", ";
+  }
+  std::cout << std::endl;
+  
+  // weight_names.resize(num_weights);
```

Comment:
Remove?

Reply:
Do you mean the weight names prints? I left them commented out because while they're using valid DLR C-API, for most models there're just too many weight nodes that the output info makes less sense. But user can still add them back easily.